srml/contracts/src/exec.rs

```diff
@@ -385,13 +388,29 @@ where
 						nested.loader.load_main(&dest_code_hash),
 						input_data
 					);
-					nested.vm
+					let output = nested.vm
 						.execute(
 							&executable,
 							nested.new_call_context(caller, value),
 							input_data,
 							gas_meter,
-						)
+						)?;
+
+					// Destroy contract if insufficient remaining balance.
+					if nested.overlay.get_balance(&dest) < nested.config.existential_deposit {
+						let parent = nested.parent
```

Comment:
I feel a bit weird about this parent construction. Passing `self` through `with_nested_context(|parent, nested|)` seems to me cleaner, did you consider going this way? If so, what are the downsides?

Reply:
Well, so the parent linking is needed to implement `is_live` because we need to check the whole execution stack. An alternative would be for the `ExecutionContext` to keep a set of all of the "live" contracts (that are being executed). But given how rarely its checked, I think linearly traversing the stack is simpler.

So given that each ExecutionContext already has a reference to its parent for that purpose, I don't see the benefit of passing `parent` into the closure as a separate param since it can be accessed from the child context already.

Reply:
Ah, yeah, I forgot about `is_live`